### PR TITLE
Setup package

### PR DESCRIPTION
**Pull request by christianlj**

### CODE REVIEW

File: lib/literature/migrations/post.ex [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      add(:meta_title, :string)
+      add(:meta_description, :string)
+      add(:custom_excerpt, :text)
+      add(:og_image, :string)
+      add(:og_title, :string)
+      add(:og_description, :string)
+      add(:twitter_image, :string)
+      add(:twitter_title, :string)
+      add(:twitter_description, :string)
+      add(:excerpt, :text)
```
Comment (samhamilton) [inline lib/literature/migrations/post.ex:24]: please rename - perhaps to html or content?

**Review by samhamilton — CHANGES_REQUESTED** [review]

File: lib/literature/components/layout_component.ex [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    </header>
+    """
+  end
+
+  def footer(assigns) do
+    ~H"""
+    <footer class="pb-4 rounded-lg md:pb-6">
+      <hr class="mb-6 border-gray-200 sm:mx-auto" />
+      <span class="block text-sm text-gray-500 sm:text-center">
+        &copy; <%= Date.utc_today().year %> <a href="https://martide.com/" class="hover:underline">Literature</a>. All Rights Reserved.
```
Comment (samhamilton) [inline lib/literature/components/layout_component.ex:56]: dont think we need this

File: lib/literature/post.ex [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    field(:og_title, :string)
+    field(:og_description, :string)
+    field(:twitter_image, Uploader.Type)
+    field(:twitter_title, :string)
+    field(:twitter_description, :string)
+    field(:excerpt, :string)
+    field(:status, :string, virtual: true)
+
+    belongs_to(:publication, Publication)
+    belongs_to(:primary_author, Author)
```
Comment (samhamilton) [inline lib/literature/post.ex:28]: post should be able to have many authors and many tags

File: lib/literature/uploader.ex [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      @extension_whitelist,
+      file.file_name
+      |> Path.extname()
+      |> String.downcase()
+    )
+  end
+
+  # Define w100 transformation:
+  def transform(:w100, _) do
+    {:convert, "-resize 10% -format webp", :webp}
```
Comment (samhamilton) [inline lib/literature/uploader.ex:30]: please verify this makes the max width 100px and not 10% of the size of the original file

File: lib/literature/post.ex (hunk 3 above)
Comment (samhamilton) [reply lib/literature/post.ex:28]: and lets rename to remove the "primary_" and go with authors and tags

**Review by technicalcapt — COMMENTED** [review]
@christianlj Code wise LGTM 👍 

I didn't test it. Let me know if we can test it from https://github.com/martide/elixir/pull/5988 🤔
(cc @samhamilton )

**Pull request merged**